### PR TITLE
Add convenience lemma weaken-↑ to substitutions library

### PR DESCRIPTION
**Pull request by dominiquedevriese**
Whenever working with the Data.Fin.Subst library, I find myself missing this lemma. Please add?

Thanks,
Dominique

### CODE REVIEW

**Comment by dominiquedevriese** [comment]
Update: I just added three more useful lemmas to this PR: wk-⊙-∷, weaken-∷ and weaken-sub′.

File: src/Data/Fin/Substitution/Lemmas.agda [hunk 1]
```diff
@@ -346,6 +346,14 @@ record Lemmas₄ {ℓ} (T : Pred ℕ ℓ) : Set ℓ where
     lookup ρ x / wk          ≡⟨ cong₂ _/_ (sym var-/) refl ⟩
     var x / ρ / wk           ∎
 
+  weaken-↑ : ∀ {k n} t {ρ : Sub T k n} → weaken t / (ρ ↑) ≡ weaken (t / ρ)
+  weaken-↑ t {ρ} = begin
```
Comment (MatthewDaggitt) [inline src/Data/Fin/Substitution/Lemmas.agda]: Please could you layout the reasoning proofs as specified in the [style guide](https://github.com/agda/agda-stdlib/blob/master/notes/style-guide.md).

File: src/Data/Fin/Substitution/Lemmas.agda [hunk 2]
```diff
@@ -464,6 +472,7 @@ record TermLemmas (T : ℕ → Set) : Set₁ where
     termSubst : TermSubst T
 
   open TermSubst termSubst
+  module T = TermSubst termSubst
```
Comment (MatthewDaggitt) [inline src/Data/Fin/Substitution/Lemmas.agda]: This module needs to be marked private.

**Comment by dominiquedevriese** [comment]
MatthewDaggit: I resolved your comments in 3 new commits.

File: src/Data/Fin/Substitution/Lemmas.agda [hunk 3]
```diff
@@ -518,3 +526,20 @@ record TermLemmas (T : ℕ → Set) : Set₁ where
...
+    lookup ρ x                  ∎)
+
+  weaken-∷ : ∀ {k n} (t₁ : T k) {t₂ : T n} {ρ : Sub T k n} → T.weaken t₁ T./ (t₂ ∷ ρ) ≡ t₁ T./ ρ
+  weaken-∷ t₁ {t₂} {ρ} = begin
+    T.weaken t₁ T./ (t₂ ∷ ρ)   ≡⟨ cong (T._/ (t₂ ∷ ρ)) (sym /-wk) ⟩
+    (t₁ T./ T.wk) T./ (t₂ ∷ ρ) ≡⟨ ⨀→/✶ ((t₂ ∷ ρ) ◅ T.wk ◅ ε) (ρ ◅ ε) (wk-⊙-∷ t₂ ρ) t₁ ⟩
+    t₁ T./ ρ                   ∎
+
+  weaken-sub′ : ∀ {n} (t₁ : T n) {t₂ : T n} → T.weaken t₁ T./ (T.sub t₂) ≡ t₁
+  weaken-sub′ t₁ {t₂} = begin
```
Comment (gallais) [inline src/Data/Fin/Substitution/Lemmas.agda]: I am not familiar with this part of the library, is `weaken-sub` already taken?

**Comment by dominiquedevriese** [comment]
My mistake, sorry, the name conflicted with a function in a different library built on top of Data.Fin.Substitution, but it makes more sense to use the unprimed name here, so this is now changed.

**Pull request merged**